src/Encamina.Enmarcha.SemanticKernel/Extensions/KernelExtensions.cs

```diff
@@ -157,6 +190,28 @@ public static async Task<int> GetKernelFunctionUsedTokensAsync(this Kernel kerne
             + GetMaxTokensFromKernelFunction(kernel, function, arguments);
     }
 
+    /// <summary>
+    /// Calculates the current total number of tokens used in generating a prompt from an inline prompt function.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> instance to work with.</param>
+    /// <param name="promptTemplate">The template of the inline prompt function.</param>
+    /// <param name="function">The inline prompt function created from the template.</param>
+    /// <param name="arguments">The arguments passed to the function.</param>
+    /// <param name="tokenLengthFunction">A function to calculate the length of a string in tokens.</param>
+    /// <param name="promptTemplateFormat">
+    /// The template format for the prompt. This must be provided if <paramref name="promptTemplateFactory"/> is not <see langword="null"/>. This parameter is optional.
+    /// </param>
+    /// <param name="promptTemplateFactory">
+    /// A <see cref="IPromptTemplateFactory"/> to interpret the prompt and its configuration into a <see cref="IPromptTemplate"/>. This parameter is optional.
+    /// </param>
+    /// <param name="cancellationToken">A cancellation token that can be used to receive notice of cancellation.</param>
+    /// <returns>The total number of tokens used, plus the maximum allowed response tokens specified in the function.</returns>
+    public static async Task<int> GetKernelFunctionUsedTokensFromPromptAsync(this Kernel kernel, string promptTemplate, KernelFunction function, KernelArguments arguments, Func<string, int> tokenLengthFunction, string promptTemplateFormat = null, IPromptTemplateFactory promptTemplateFactory = null, CancellationToken cancellationToken = default)
```

Comment:
You have to take into consideration the warnings generated on the new code by the integration process. Please check the following warning: Split this 346 characters long line (which is greater than 200 authorized). (https://rules.sonarsource.com/csharp/RSPEC-103)

Reply:
Perfect, I mark it as solved. We should update the ruleset.